trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/model/ConfigurationQueryParameters.java

```diff
@@ -23,9 +21,9 @@
 public interface ConfigurationQueryParameters {
 
     /**
-     * @return The query parameter map.
+     * @return The query parameters.
      */
     @NonNull
-    @Schema(required = true, description = "Parameters as specified in corresponding ConfigurationTypeDescriptor. Use key `path` for file URI string values.")
-    Map<String, Object> getParameters();
+    @Schema(required = true, description = "Parameters as JSON object as specified in the schema of the corresponding ConfigurationTypeDescriptor or as a Map<String, Object>. Use key `path` for file URI string values.")
```

Comment:
Yes, this file is only used for swagger documentation.

An Object resolves to a JsonObject in generated swagger documentation as far as I understand. 

The client will have to put a map<String, Object> where the key is `parameters` and the value the actual input .JSON object for this configuration (see below for an example). I kept the parameters map because it is consistent to all the other endpoints. What do you suggest to do it differently? I mean in general in the implementation as well as for the swagger documention.

```json
{
    "parameters": {
        "name": "my InAndOut",
        "description" : "my description",
        "specifiers" : [
            {
            "label":"latency",
            "inRegex":"(\\S*)_entry",
            "outRegex":"(\\S*)_exit",
            "contextInRegex":"(\\S*)_entry",
            "contextOutRegex":"(\\S*)_exit",
            "classifier":"CPU"
            }
        ]
    },
    "typeId": "org.eclipse.tracecompass.incubator.internal.inandout.core.config"
}
```

Reply:
I guess what I'm understanding is that at TSP protocol (swagger) level, we don't need to make the difference, the example you give fits both. So we could just describe it here by modifying the original to add 'as specified in the schema of...'. It's not clear to me how the sender would differentiate between sending a JSON object or a Map, or how it would look different over the TSP, so I'm not sure about describing it in the protocol as a choice.

Reply:
A JSON object in swagger boils done to a map if it is deserialized as a map with default deserializer.